chall08/vsempere.py

```diff
@@ -0,0 +1,124 @@
+#! /usr/local/bin/python3
+
+import sys
+
+def process_user_input(user_input):
+    if (not isinstance(user_input, list)):
+        return None, None, None, 'Invalid input.';
+
+    map_line_size = -1
+    map_read = 0
+    moves_read = 0
+    map = ""
+    for line in user_input:
+        line = line.rstrip()
+        if (map_read == 0):
+            if (len(line) == 0):
+                map_read = 1
+            else:
+                if (map_line_size < 0):
+                    map_line_size = len(line)
+                elif (map_line_size != len(line)):
+                    return None, None, None, 'Invalid input.'
+                map = map + line
+        else:
+            if (moves_read == 0):
+                moves = line
+                moves_read = 1
+            else:
+                return None, None, None, 'Invalid input.'
+    return map, map_line_size, moves, None
+
+def check_user_input(map, map_line_size, moves):
+    if (map is None or map_line_size is None or moves is None):
+        return 0
+
+    rows = len(map) / map_line_size
+    row = 0
+    col = 0
+    exits = 0
+    robots = 0
+    for map_cell in map:
+        if (row == 0  or row == (rows-1) or col == 0 or col == (map_line_size -1)):
+            if (not map_cell in "+E"):
+                return 0
+            if (map_cell == "E"):
+                exits = exits + 1
+        else:
+            if (not map_cell in "0*M"):
+                return 0
+            if (map_cell == 'M'):
+                robots = robots + 1
+        col = col + 1
+        if (col == map_line_size):
+            col = 0
+            row = row + 1
+    
+    if (row != rows or robots != 1 or exits != 1):
+        return 0
+    
+    for move in moves:
+        if (not move in "IKNSEW"):
+            return 0
+    return 1
+
+def get_robot_pos(map):
+    pos = 0
+    for cell in map:
+        if cell=='M':
+            return pos
+        pos = pos + 1
```

Comment:
You already have a builtin for that ;) 
```suggestion
    return map.index("M")
```